### PR TITLE
Optimize most important parts with ARM NEON SIMD

### PR DESCRIPTION
**Pull request by danlark1**
First part, updated most UTF8, hashing, memory and codecs. Except
utf8lower and upper, maybe a little later.

That includes huge amount of research with movemask dealing. Exact
details and blog post TBD.

Partially deals with #37005

### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added numerous NEON accelerated paths for main libraries

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by qoega** [comment]
I added 'force tests' lable. If aarch64 build will be ready perf tests will be run even if there will be failures in other checks

**Comment by alexey-milovidov** [comment]
> Stateless tests (thread, actions) [1/3] — fail: 2, passed: 1396, skipped: 11 

Bug in TSan: https://github.com/google/sanitizers/issues/1540

File: tests/queries/0_stateless/01375_compact_parts_codecs.reference [hunk 1]
```diff
@@ -4,6 +4,6 @@
 11965	11890
 499500	499500	999
 499500	499500	999
-5858	11890
+5857	11890
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/01375_compact_parts_codecs.reference:7]: Ok. ZSTD update, better compression.

File: src/Common/HashTable/Hash.h [hunk 2]
```diff
@@ -339,6 +352,21 @@ struct UInt256HashCRC32
     }
 };
 
+#elif defined(__aarch64__) && defined(__ARM_FEATURE_CRC32)
+
+struct UInt256HashCRC32
```
Comment (alexey-milovidov) [inline src/Common/HashTable/Hash.h:357]: Will lead to incompatibility of different versions on cluster, I will add a note.

**Review by alexey-milovidov — APPROVED** [review]

File: cmake/cpu_features.cmake [hunk 3]
```diff
@@ -29,7 +29,7 @@ if (ARCH_NATIVE)
     set (COMPILER_FLAGS "${COMPILER_FLAGS} -march=native")
 
 elseif (ARCH_AARCH64)
-    set (COMPILER_FLAGS "${COMPILER_FLAGS} -march=armv8-a+crc")
+    set (COMPILER_FLAGS "${COMPILER_FLAGS} -march=armv8-a+crc+simd+crypto+dotprod+ssbs")
```
Comment (alexey-milovidov) [inline cmake/cpu_features.cmake:32]: What CPUs will be not compatible?
There are some Cavium and APM servers from 2013...

Comment (alexey-milovidov) [reply cmake/cpu_features.cmake:32]: Raspberry Pi?

Comment (danlark1) [reply cmake/cpu_features.cmake:32]: Yes, Cortex A-72 does not support ssbs, everything else is supported. We found quite useful ssbs feature when kernel is enabled like for Graviton, however, not that significant, if somebody complains, we can drop it

**Pull request merged**